lib/Validation.groovy

```diff
@@ -0,0 +1,115 @@
+import groovy.json.JsonSlurper
+import groovy.util.logging.Log
+
+class Validation {
+
+    /*
+    * Function to loop over all parameters defined in schema and check
+    * whether the given paremeters adhere to the specificiations
+    */
+    private static void validateParameters(params, json_schema, log, workflow){
+
+        def json = new File(json_schema).text
+        def Map json_params = (Map) new JsonSlurper().parseText(json).get('definitions')
+        def specified_param_keys = params.keySet()
+        def nf_params = ['profile', 'config', 'c', 'C', 'syslog', 'd', 'dockerize', 
+                        'bg', 'h', 'log', 'quiet', 'q', 'v', 'version']
+        def valid_params = []
+        def expected_params = []
+        def blacklist  = ['hostnames'] // ignored parameters
+        
+
+        // Loop over all parameters in schema and compare to given parameters
+        for (group in json_params){
+            for (p in group.value['properties']){
+                if (!blacklist.contains(p.key)){
+                    valid_params.push(validateParamPair(params[p.key], p, log))
+                    expected_params.push(p.key)
+                }
+            }
+        }
+
+        // Exit if any invalid params where found
+        if (valid_params.contains(false)){
+            System.exit(0)
+        }
+
+        // Check for nextflow core params and unexpected params
+        for (specified_param in specified_param_keys){
+            // nextflow params
+            if (nf_params.contains(specified_param)){
+                log.error "ERROR: You have overwritten the core Nextflow parameter -${specified_param} with --${specified_param}!"
+                System.exit(0)
+            }
+            // unexpected params
+            if (!expected_params.contains(specified_param)){
+                log.warn "Unexpected parameter specified: ${specified_param}"               
```

Comment:
See https://github.com/nf-core/cageseq/pull/38#discussion_r544597458 for ideas about when we might call `System.exit()` here..